### PR TITLE
Add command to resist movement for use while shooting

### PR DESCRIPTION
**Pull request by WasabiFan**

### CODE REVIEW

File: Competition2017/src/competition/subsystems/drive/commands/ResistMovementCommand.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import competition.operator_interface.OperatorInterface;
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+import xbot.common.math.PIDFactory;
+import xbot.common.math.PIDManager;
+import xbot.common.properties.XPropertyManager;
+
+public class ResistMovementCommand extends BaseDriveCommand {
+    PIDManager leftPid;
+    PIDManager rightPid;
```
Comment (imathno) [inline Competition2017/src/competition/subsystems/drive/commands/ResistMovementCommand.java]: Why make this public?

File: Competition2017/src/competition/subsystems/drive/commands/ResistMovementCommand.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package competition.subsystems.drive.commands;
+
+import com.google.inject.Inject;
+
+import competition.operator_interface.OperatorInterface;
```
Comment (imathno) [inline Competition2017/src/competition/subsystems/drive/commands/ResistMovementCommand.java]: Remove unused import

File: Competition2017/src/competition/subsystems/drive/commands/ResistMovementCommand.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
```
Comment (imathno) [inline Competition2017/src/competition/subsystems/drive/commands/ResistMovementCommand.java]: Unused import

File: Competition2017/src/competition/subsystems/drive/commands/ResistMovementCommand.java (hunk 1 above)
Comment (WasabiFan) [reply Competition2017/src/competition/subsystems/drive/commands/ResistMovementCommand.java]: Fixed; good call! It's worth noting that Java's default accessibility is "package protected," meaning that it's accessible to other classes of the same package and inaccessible elsewhere. "package protected" is almost always not what you want.

File: Competition2017/src/competition/subsystems/drive/commands/ResistMovementCommand.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    private double targetRightPosition;
+    
+    @Inject
+    public ResistMovementCommand(
+            DriveSubsystem driveSubsystem,
+            PIDFactory pidFactory,
+            XPropertyManager propMan) {
+        super(driveSubsystem);
+        
+        final String sharedPidKey = "Resist movement single-side";
```
Comment (aschokking) [inline Competition2017/src/competition/subsystems/drive/commands/ResistMovementCommand.java]: Probably can get away with just `"Resist movement"`?

**Review by aschokking — APPROVED** [review]

**Review by imathno — APPROVED** [review]

**Pull request merged**